### ISSUE TITLE
OCTeam view to remove user account 

### ISSUE THREAD

**Issue by kojoty** [opencaching/opencaching-pl#2226]
Until now there is no simple way to remove user account on user request and site admins do this manually.

I think we should change this.

First let's add the option for OC-Team:

1. search uses account by email address (user should request account deletion by email assigned to account (to be sure who is requester) 

2. then user account can be removed with options:
- account after the operation will be locked 
- account username will be anonymized (because of personal data) - removed account prefix will be provided by node configuration
- user email will be removed
- additional user settings stored in DB will be removed - there is no sense to store for example list of watched caches for removed accounts
- information about the OC-Team member who remove the account will be saved

3. there is no way to undelete "removed" account

Next I'm going to implement the solution to allow user to delete their account - but this will be defined in another issue.

**Comment by andrixnet** [comment]
With all GDPR implications as well. (right to be forgotten, etc)

**Comment by andrixnet** [comment]
IMO this functionality should exists two-fold: 
- user to "delete/deactivate own account forever"
- admin to "delete/deactivate selected account"

With all due warnings and confirmations.

**Comment by kojoty** [comment]
> IMO this functionality should exists two-fold:
> 
> * user to "delete/deactivate own account forever"
> * admin to "delete/deactivate selected account"
> 
> With all due warnings and confirmations.

Yes - I'm want to implement the second option first and the first later - the first option is a little bit more complicated, but most of the code will be reused.
